src/box/replication.cc

```diff
@@ -206,6 +210,71 @@ replicaset_set_sync_quorum(void)
 	replication_sync_quorum_auto = replicaset.applier.connected;
 }
 
+/**
+ * Deactivates all expired anonymous replicas. They are expected to be
+ * deleted on deactivation as orphan.
+ * Returns lowest time-to-live among remaining anonymous replicas via
+ * `lowest_ttl`. If there are no anonymous replicas left or the function has
+ * failed to deactivate a replica, `replication_anon_gc_timeout` is returned.
+ */
+static void
+replication_anon_gc(double *lowest_ttl)
+{
+	double timeout = replication_anon_gc_timeout;
+	while (true) {
+		*lowest_ttl = timeout;
+		struct replica *expired_replica = NULL;
+		/*
+		 * Search for the first expired replica. Note that we cannot
+		 * just deactivate it and continue the iteration because a yield
+		 * will happen and the replicaset can be changed significantly.
+		 */
+		replicaset_foreach(replica) {
+			if (!replica->anon || replica_has_connections(replica))
+				continue;
+			double last_seen = relay_last_row_time(replica->relay);
+			double unused = clock_monotonic() - last_seen;
```

Comment:
Relay's `last_row_time` is set as `ev_monotonic_now(loop())`. Lets use the same timer here, not `clock_monotonic()`. The difference is that the latter is an actual syscall, while the former isn't. Also libev timer can be a bit behind the real monotonic clock, because is collected only once per event loop iteration.

Reply:
Fiber works extremely rare, so I used `clock_monotonic` here intentionally - to be more precise at a price of system call. But it doesn't really matter for me - let's use libev clock for consistency.
Replaced with `ev_monotonic_now(loop())`.